### PR TITLE
Add compat data for Request (Fetch API)

### PR DESCRIPTION
**Pull request by maboa**
[https://developer.mozilla.org/en-US/docs/Web/API/Request](https://developer.mozilla.org/en-US/docs/Web/API/Request)

Note the `Body` methods and properties linked to from `Request` are here: [https://github.com/mdn/browser-compat-data/pull/916](https://github.com/mdn/browser-compat-data/pull/916)

Not sure what to do about the following linting error:

```File : /Users/markb/Sites/browser-compat-data/api/Request.json
  Style – Error on line #397
    Actual:                 "version_added": "15"
    Expected:               "version_added": true
```

### CODE REVIEW

File: api/Request.json [hunk 1]
```diff
@@ -0,0 +1,1425 @@
...
+            "version_added": false
+          }
+        },
+        "status": {
+          "experimental": true,
+          "standard_track": true,
+          "deprecated": false
+        }
+      },
+      "request_body_formdata": {
```
Comment (Elchi3) [inline api/Request.json]: Remove this, the data is here https://github.com/mdn/browser-compat-data/blob/master/api/Body.json#L380

File: api/Request.json [hunk 2]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            }
+          }
+        }
+      },
+      "request_integrity": {
```
Comment (Elchi3) [inline api/Request.json]: Remove this, redundant with `integrity` data below.

File: api/Request.json [hunk 3]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            }
+          }
+        }
+      },
+      "request_redirect": {
```
Comment (Elchi3) [inline api/Request.json]: Remove this, redundant with `redirect` data below.

File: api/Request.json [hunk 4]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled

File: api/Request.json [hunk 5]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 6]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            }
+          }
+        }
+      },
+      "accept_readablestream_body": {
```
Comment (Elchi3) [inline api/Request.json]: Remove this. Redundant with the entry under the constructor.

File: api/Request.json [hunk 7]
```diff
@@ -0,0 +1,1425 @@
...
+            }
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "accept_readablestream": {
```
Comment (Elchi3) [inline api/Request.json]: needs to be nested under the constructor data.

File: api/Request.json [hunk 8]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": "10.3"
+            }
+          }
+        }
+      },
+      "navigate_mode": {
```
Comment (Elchi3) [inline api/Request.json]: needs to be nested under the constructor data.

File: api/Request.json [hunk 9]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": "10.1"
+            },
+            "safari_ios": {
+              "version_added": "10.3"
+            }
+          }
+        }
+      },
+      "referrer_init": {
```
Comment (Elchi3) [inline api/Request.json]: needs to be nested under the constructor data.

File: api/Request.json [hunk 10]
```diff
@@ -0,0 +1,1425 @@
...
+            "safari": {
+              "version_added": "10.1"
+            },
+            "safari_ios": {
+              "version_added": "10.3"
+            }
+          }
+        }
+      },
+      "cross_origin_stripped": {
```
Comment (Elchi3) [inline api/Request.json]: needs to be nested under the constructor data.

File: api/Request.json [hunk 11]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 12]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled

File: api/Request.json [hunk 13]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 14]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 15]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled

File: api/Request.json [hunk 16]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 17]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 18]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled

File: api/Request.json [hunk 19]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 20]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 21]
```diff
@@ -0,0 +1,1425 @@
...
+                  }
+                ]
+              }
+            ],
+            "chrome_android": {
+              "version_added": "42",
+              "version_removed": "46"
```
Comment (Elchi3) [inline api/Request.json]: Not sure where this is coming from. Removed in 46?

File: api/Request.json [hunk 22]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 23]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 24]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled

File: api/Request.json [hunk 25]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 26]
```diff
@@ -0,0 +1,1425 @@
...
+            }
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "referrerPolicy": {
```
Comment (Elchi3) [inline api/Request.json]: referrerPolicy data is added twice. This might explain the travis failure?!

File: api/Request.json [hunk 27]
```diff
@@ -0,0 +1,1425 @@
...
+            "chrome": [
+              {
+                "version_added": "42"
+              },
+              {
+                "version_added": "41",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json [hunk 28]
```diff
@@ -0,0 +1,1425 @@
...
+                  }
+                ]
+              }
+            ],
+            "chrome_android": {
+              "version_added": "42",
+              "version_removed": "46"
```
Comment (Elchi3) [inline api/Request.json]: I think version_removed is wrong here.

File: api/Request.json [hunk 29]
```diff
@@ -0,0 +1,1425 @@
...
+            "firefox": [
+              {
+                "version_added": "39"
+              },
+              {
+                "version_added": "34",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: dom.fetch.enabled.

File: api/Request.json [hunk 30]
```diff
@@ -0,0 +1,1425 @@
...
+            "opera": [
+              {
+                "version_added": "29"
+              },
+              {
+                "version_added": "28",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": ""
```
Comment (Elchi3) [inline api/Request.json]: "Experimental Web Platform Features"

File: api/Request.json (hunk 1 above)
Comment (maboa) [reply api/Request.json]: 👍

File: api/Request.json (hunk 2 above)
Comment (maboa) [reply api/Request.json]: 👍

File: api/Request.json (hunk 3 above)
Comment (maboa) [reply api/Request.json]: 👍

File: api/Request.json (hunk 6 above)
Comment (maboa) [reply api/Request.json]: 👍

File: api/Request.json (hunk 21 above)
Comment (maboa) [reply api/Request.json]: Changed to : 
```
            "chrome_android": {
              "version_added": false
            },
```
As specified here https://developer.mozilla.org/en-US/docs/Web/API/Request/referrer

File: api/Request.json (hunk 26 above)
Comment (maboa) [reply api/Request.json]: 👍

File: api/Request.json (hunk 28 above)
Comment (maboa) [reply api/Request.json]: changed to:

```
            "chrome_android": {
              "version_added": false
            },
```
As specified here : https://developer.mozilla.org/en-US/docs/Web/API/Request/clone

**Comment by maboa** [comment]
Edited. However the Travis error remains:

```
Style – Error on line #215
Actual:                 "version_added": "15"
Expected:               "version_added": true```

This is for `edge`.

**Comment by Elchi3** [comment]
This is the nesting right now, which fails because there are two navigate_mode entries:

- Request
  - __compat
  - Reqeust
  - accept_readablestream
  - navigate_mode
  - referrer_init
  - cross_origin_stripped
      - cache
      - only-if-cached
  - context
  - credentials
  - headers
  - integrity
  - method
  - mode
  - navigate_mode
  - redirect
  - referrer
  - referrerPolicy
  - url
  - clone

<hr>

This is what we need:

- Request
  - __compat
  - Request
    - accept_readablestream
    - navigate_mode
    - referrer_init
    - cross_origin_stripped
   - cache
     - only-if-cached
  - context
  - credentials
  - headers
  - integrity
  - method
  - mode
    - navigate_mode
  - redirect
  - referrer
  - referrerPolicy
  - url
  - clone

File: api/Request.json [hunk 31]
```diff
@@ -0,0 +1,1193 @@
...
+            "edge": {
+              "version_added": null
+            },
+            "edge_mobile": {
+              "version_added": null
+            },
+            "firefox": {
+              "version_added": "39",
+              "version_removed": "42",
+              "notes":"see <a href='https://bugzil.la/1188062'>bug 1188062</a> for more information."
```
Comment (Elchi3) [inline api/Request.json]: Add a space here to fix a style failure:

```diff
-"notes":"see .."
+"notes": "see .."
```

**Comment by maboa** [comment]
Thanks. Hopefully this is all fixed now :)

**Pull request merged**